Test/PublishPowerShellScript.Tests.ps1

```diff
@@ -0,0 +1,433 @@
+    {
+        $To = Join-Path -Path $script:testRoot -ChildPath $To
+    }
+
+    Join-Path -Path $To -ChildPath "$($testScriptName).*.*.*$($WithPrerelease).nupkg" | Should -Exist
+}
+
+function ThenManifest
+{
+    param(
+        [String]$ManifestPath = (Join-Path -Path $testRoot -ChildPath "$($testScriptName)\$($testScriptName).ps1"),
+
+        [String]$HasPrerelease
+    )
+
+    Test-ScriptFileInfo -Path $ManifestPath | Should -Not -BeNullOrEmpty
+    Get-Content -Raw -Path $ManifestPath | Should -Match ".VERSION 1.2.3-$($HasPrerelease)"
+}
+
+function ThenSucceeded
+{
+    $script:failed | Should -BeFalse
+    Get-Error | Should -BeNullOrEmpty
+}
+
+function Get-Error
+{
+    [CmdletBinding()]
+    param(
+    )
+
+    # PackageManagement and PowerShellGet leave handled errors in Global:Error so we need to filter those errors out.
+    $Global:Error |
+        Where-Object 'ScriptStackTrace' -NotMatch '\bPowerShellGet\b' -ErrorAction Ignore |
+        Where-Object 'ScriptStackTrace' -NotMatch '\bPackageManagement\b' -ErrorAction Ignore
+}
+
+Describe 'PublishPowerShellScript.when publishing to repository that already exists' {
+    AfterEach { Reset }
+    It 'should publish the script wihtout registering the repository' {
+        Init
+        GivenRepository 'R1'
+        WhenPublishing -ToRepo 'R1'
+        ThenSucceeded
+        ThenScriptPublished -To 'R1'
+        ThenRepository 'R1' -Exists -NotRegistered
+    }
+}
+
+Describe 'PublishPowerShellScript.when publishing to repository that does not exist' {
+    AfterEach { Reset }
+    It 'should fail' {
+        Init
+        WhenPublishing -ToRepo 'R2' -ErrorAction Silently
+        ThenFailed 'a repository with that name doesn''t exist'
+        ThenRepository 'R2' -NotExists -NotUnregistered
+    }
+}
+
+Describe 'PublishPowerShellScript.when publishing prerelease script' {
+    AfterEach { Reset }
+    It 'should succeed' {
+        Init
+        WhenPublishing -WithPrerelease 'beta1'
+        ThenSucceeded
+        ThenScriptPublished -WithPrerelease '-beta1'
+        ThenManifest -HasPrerelease 'beta1'
+    }
+}
+
+Describe 'PublishPowerShellScript.when publishing with no repository name or repository location' {
+    AfterEach { Reset }
+    It 'should publish to build output directory' {
+        Init
+        WhenPublishing
+        ThenSucceeded
+        ThenRepository 'Whiskey*' -NotExists
+        ThenScriptPublished -To $context.OutputDirectory
+    }
+}
+
+Describe 'PublishPowerShellScript.when given an API key' {
+    AfterEach { Reset }
+    It 'should fail' {
+        Init
+        Mock -CommandName 'Publish-Script' `
+             -ModuleName 'Whiskey' `
+             -ParameterFilter { $NuGetApiKey -eq 'API6' } `
+             -Verifiable
+        WhenPublishing -WithApiKey 'API6'
+        ThenSucceeded
+        Assert-VerifiableMock
+    }
+}
+
+Describe 'PublishPowerShellScript.when path parameter is not included' {
+    AfterEach { Reset }

```

Comment:
Add a `ThenScriptNotPublished` assertion to each test that asserts failure. The `ThenScriptNotPublished` should check that no *.nupkg files were published to the output directory.